aim/api/resource.py

```diff
@@ -109,6 +109,32 @@ def make_serializable(key, attr):
             json.dumps(serializable, sort_keys=True))).hexdigest(), 16)
 
     def user_equal(self, other):
+        # REVISIT: this is a bit ugly, but will work for the current
+        #          resource types. If the attribute is a list of scalar
+        #          types, then we can just use equivalency directly. If
+        #          they are non-scalar (e.g. dictionaries), then we just
+        #          check if the new one is a subset of the existing one.
+        #          This was added because there are default values present
+        #          in the model representation, and if an update leaves
+        #          these fields out, then we don't want the equivalency
+        #          check to fail. This solution will not work if there is
+        #          a resource that uses non-scalar types in the list
+        #          that are allowed to have some elements absent.
+        def attributes_equivalent(superset, subset):
```

Comment:
Per my previous comment, I don't think this is really needed.

Reply:
see my comment above